docs/source/working/data.rst

```diff
@@ -0,0 +1,148 @@
+
+    class NewData(Data)
+        """A new data type that wraps a single value."""
+
+        attribute_name = 'value'
+
+        def __init__(self, **kwargs)
+            value = kwargs.pop(self.attribute_name)
+            super(NewData, self).__init__(**kwargs)
+
+.. warning::
+
+    For the class to function properly, the signature of the constructor **cannot be changed** and the constructor of the parent class **has to be called**.
+
+Before calling the construtor of the base class, we have to remove the ``value`` keyword from the keyword arguments ``kwargs``, because the base class will not expect it and will raise an exception if left in the keyword arguments.
+The final step is to actually *store* the value that is passed by the caller of the constructor.
+The previous snippet removes it from the ``kwargs`` but then does not do anything with it.
+A new node has two locations to permanently store any of its properties:
+
+* the database
+* the file repository
+
+The section on :ref:`design guidelines<working_data_design_guidelines>` will go into more detail what the advantages and disadvantages of each option are and when to use which.
+For now, since we are storing only a single value, the easiest and best option is to use the database.
+Each node has *attributes* that can store any key-value pair, as long as the value is JSON serializable.
+We can use these attributes to store the ``value`` of our ``NewData`` data type:
+
+.. code:: python
+
+    from aiida.orm import Data
+
+    class NewData(Data)
+        """A new data type that wraps a single value."""
+
+        attribute_name = 'value'
+
+        def __init__(self, **kwargs)
+            value = kwargs.pop(self.attribute_name)
+            super(NewData, self).__init__(**kwargs)
+            self.set_attribute(self.attribute_name, value)
+
+By adding the value to the node's attributes, after the node itself has been stored, it will be permanently recorded in the database.
+Creating an instance of the ``NewData`` data type and storing it is now very easy:
+
+.. code:: python
+
+    node = Data(value=5)
+    node.store()
+
+By storing the ``value`` in the attributes of the node instance, we ensure that that ``value`` can be retrieved even when the node is reloaded at a later point in time.
+
+Besides making sure that the content of a data node is stored in the database or file repository, the data type class can also provide useful methods for users to retrieve that data.
+For example, with the current state of the ``NewData`` class in the last example, if a user were to load a stored ``NewData`` node and wanted to retrieve the ``value`` that it contains, they would have to do something like:
+
+.. code:: python
+
+    node = load_node(<IDENTIFIER>)
+    node.get_attribute(NewData.attribute_name)
+
+In other words, the user of the ``NewData`` class needs to know that the ``value`` is stored as an attribute with a name given by the class-attribute ``attribute_name``.
```

Comment:
If I understand correctly you suggest to remove this and the following line entirely. I actually think with this trivial example these lines are necessary. In this case doing `node.get_attribute('v alue')` is not _that_ much more difficult or weird then doing `node.value`, so I fear that by not being very explicit this will be lost on some readers.

Reply:
Ok, fine